common/repl/repl_wrapper.cpp

```diff
@@ -289,6 +291,7 @@ REPL::Config load_repl_config(const std::string& username,
       // do nothing
     }
   }
+  loaded_config.iso_path = file_util::get_iso_dir_for_game(game_version).string();
   loaded_config.temp_nrepl_port = nrepl_port;
   return loaded_config;
```

Comment:
this is odd behaviour -- you support loading the `isoPath` from the json file, but then set it here.  This means that if someone wanted to set it in the file, their value gets immediately clobbered.  This also has the problem of this line of code only executing if the user _doesnt_ have a `repl-config.json` (see the return above).

I think this line should be removed.  in `main.cpp` you should set this field on the struct there if `iso_path_override` is provided.

This also means that this value would only be set when it's actually set via the CLI flag (right now `iso_path` is always non-empty if you don't have a `repl-config.json`, so it always executes your new code path in `MakeSystem.cpp`)

Reply:
Also reminder to copy in this fix to resolve the CI issue https://github.com/open-goal/jak-project/commit/dc957f60fadcadc417676c94a6e90fd24651f955

Reply:
Wanna make sure I have this right before committing.

Here I removed the line you mentioned.
![image](https://github.com/user-attachments/assets/ffb6c4b6-b380-4c6b-81f7-a57694a60f90)

Then here, I have to move our `iso_path_override` check after the instantiation of `repl_config`, and we set the `repl_config.iso_path`. 
![image](https://github.com/user-attachments/assets/d509d54e-cb6f-4e77-9064-e6a786c3aabc)

So if a user has a repl_config _and_ a CLI override, the override takes priority. If they have a repl_config with no override, the value in the config applies.

Is there anything I need to do to handle a situation where the config has no value and we don't have an override?

Reply:
> Is there anything I need to do to handle a situation where the config has no value and we don't have an override?
There shouldn't be, you're later code checks if `iso_path` is empty, and if so you continue to do what `MakeSystem.cpp` was already doing prior to this PR.

Reply:
Ok, I'll test later tonight/tomorrow.